cranelift/codegen/src/isa/x64/lower/isle.rs

```diff
@@ -546,27 +546,14 @@ impl Context for IsleContext<'_, '_, MInst, Flags, IsaFlags, 6> {
         Imm8Gpr::new(Imm8Reg::Imm8 { imm }).unwrap()
     }
 
-    fn is_gpr_type(&mut self, ty: Type) -> Option<Type> {
-        if is_int_or_ref_ty(ty) || ty == I128 || ty == B128 {
-            Some(ty)
-        } else {
-            None
-        }
-    }
-
-    #[inline]
-    fn is_xmm_type(&mut self, ty: Type) -> Option<Type> {
-        if ty == F32 || ty == F64 || (ty.is_vector() && ty.bits() == 128) {
-            Some(ty)
-        } else {
-            None
-        }
-    }
-
     #[inline]
-    fn is_single_register_type(&mut self, ty: Type) -> Option<Type> {
-        if ty != I128 {
-            Some(ty)
+    fn type_register_class(&mut self, ty: Type) -> Option<RegisterClass> {
+        if is_int_or_ref_ty(ty) || ty == I128 || ty == B128 {
+            Some(RegisterClass::Gpr {
+                single_register: ty != I128,
```

Comment:
No. As currently implemented, they would be either 128 `0`s, or 128 `1`s.

Reply:
Then the expression for `single_register` should be `ty != I128 && ty != B128`, right?

Reply:
The previous implementation of `is_single_register_type` only checked that the type wasn't equal to `I128`, so I opted to preserve the existing behavior.
https://github.com/bytecodealliance/wasmtime/blob/3a2b32bf4df83ab3d624f8c187116d02adb2c69c/cranelift/codegen/src/isa/x64/lower/isle.rs#L567-L573

Reply:
I have concerns about the previous implementation then. :sweat_smile: I'm guessing this means we've never handed a B128 to the x64 lowering rules that used this check. And that seems... not unlikely, I suppose?

Reply:
Sure, but let's keep two things in mind while reviewing this PR: 1) we're not trying to fix existing bugs, only resolve overlaps, 2) we're talking about removing all the boolean types in the short-term.

Reply:
Yep, I agree with both of your points. I think it's still worth noting existing bugs during any review, and in general opening issues to track them if we decide not to fix them immediately. But in this specific case, you're right, it doesn't matter since we're going to remove the boolean types.